village/src/components/Smurf.js

```diff
@@ -1,13 +1,24 @@
-import React from 'react';
+import React, { Component }from 'react';
 
-const Smurf = props => {
-  return (
+class Smurf extends Component {
+
+  buttonGargamel = e =>{
+    e.preventDefault();
+    this.props.buttonGargamel(this.state)
```

Comment:
make sure that this prop is making its way all the way down to the Smurf component. Also, do you have state here?